packages/marketo-form/server/api-routes.ts

```diff
@@ -39,9 +58,35 @@ async function getForm(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-async function submitForm(req: NextApiRequest, res: NextApiResponse) {
-  // Don't submit forms to the Marketo API when using E2E tests.
+function isE2ETest(req: NextApiRequest): boolean {
+  // running e2e tests locally
   if (process.env.E2E_TESTS === 'true') {
+    return true
+  }
+
+  // running e2e tests against a Vercel preview
+  if (
+    'input' in req.body &&
+    req.body.input.length > 0 &&
+    req.body.input[0].leadFormFields.email === 'daniela.rod@example.com'
```

Comment:
I'm guessing this is an arbitrary fixture email that's used? 😄

Reply:
It is! I'm not in love with this solution, but it's the best I could come up with given the constraints that we want preview deployments to be able to submit to Marketo, and we want E2E tests to run against preview deployments.

I could probably check the UA for `puppeteer` or something, but I'll explore that later.